modules/installation-upi-machines-iso.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_bare_metal/installing-bare-metal.adoc
+
+[id="installation-upi-machines-iso-{context}"]
+= Creating {op-system-first} machines using an ISO image
+
+Before you install a user provisioned infrastructure (UPI) cluster on bare
+metal, you need to create {op-system} machines for it to use. You can use an
+ISO image to create the machines.
+
+.Prerequisites
+
+* Obtain the Ignition config files for your cluster.
+* Have access to an HTTP server that you can access from your computer and that
+the machines that you create can access.
+
+.Procedure
+
+////
+This will work for GA.
+. Obtain the {op-system} images from the
+link:https://access.redhat.com/downloads[Product Downloads] page on the Red
+Hat customer portal.
+////
+
+. Upload the control plane, compute, and bootstrap Ignition config files that the
+installation program created to your HTTP server. Note the URLs of these files.
+
+. Obtain the {op-system} ISO image:
++
+----
+$ curl -J -L -O https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/410.8.20190412.1/rhcos-410.8.20190412.1-installer.iso
+----
+
+. Use the ISO to install to start the {op-system} installation. Use one of the following
+installation options:
+** Burn the ISO image to a disk and boot it directly.
+** Use ISO redirection via a LOM interface.
+
+. After the instance boots, press the `TAB` or `E` key to edit the kernel command line.
+. Add the parameters to the kernel command line:
++
+----
+coreos.inst=yes
+coreos.inst.install_dev=sda <1>
+coreos.inst.image_url=https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/410.8.20190412.1/rhcos-410.8.20190412.1-installer.iso <2>
```

Comment:
The intended workflow is to pass the metal image (`rhcos-<version>.<timestamp>.<respin>-metal-<uefi/bios>.raw.gz`). Passing the ISO to `coreos.inst.image_url` will most likely fail (haven't personally tested, but `coreos-installer` is just running essentially `zcat $file | dd`).

Reply:
@arithx, assuming that you're booting from the ISO, do you need to pass this image?

Reply:
Yes, the variable will have to be specified. If it's not specified on the kernel cmdline then it will be prompted for inside the installer.

Reply:
@arithx, how do I get that value?

Reply:
In your current line just change `installer.iso` to `metal-uefi.raw` or `metal-bios.raw` (NOTE: I believe the file paths for these images are changed in the recent versions of the pipeline to be `raw.gz` instead of just `raw`)

Reply:
Thank you, @arithx!